src/app/hero-detail.component.ts

```diff
@@ -0,0 +1,21 @@
+import { Component, Input } from '@angular/core';
+import { Hero } from './hero';
+
+@Component ({
+ selector: 'my-hero-detail',
+ template: `
+  <div *ngIf="hero">
+    <h2>{{hero.name}} details!</h2>
+    <div><label>id: </label>{{hero.id}}</div>
+    <div>
+      <label>name: </label>
+      <input [(ngModel)]="hero.name" placeholder="name"/>
```

Comment:
It's common with `ngModel` to use both `input`/`output` bindings together. This is somewhat atypical, however, so in non-ngModel cases you'll likely be separating the two, as you've done in the [ngFor](https://github.com/eshans/ng2-tour-of-heroes/pull/3/files#diff-5c26d2c8f8838f32ace87fdecd6344eaR22) above.